fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/LogicalUniqueExpression.java

```diff
@@ -0,0 +1,123 @@
+import com.apple.foundationdb.record.query.plan.cascades.TranslationMap;
+import com.apple.foundationdb.record.query.plan.cascades.explain.InternalPlannerGraphRewritable;
+import com.apple.foundationdb.record.query.plan.cascades.explain.NodeInfo;
+import com.apple.foundationdb.record.query.plan.cascades.explain.PlannerGraph;
+import com.apple.foundationdb.record.query.plan.cascades.values.Value;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A relational planner expression representing a stream of unique records. This expression has a single child which
+ * is also a {@link RelationalExpression}.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class LogicalUniqueExpression implements RelationalExpressionWithChildren, InternalPlannerGraphRewritable {
+    @Nonnull
+    private final Quantifier inner;
+
+    public LogicalUniqueExpression(@Nonnull ExpressionRef<RelationalExpression> innerRef) {
+        this(Quantifier.forEach(innerRef));
+    }
+
+    public LogicalUniqueExpression(@Nonnull Quantifier inner) {
+        this.inner = inner;
+    }
+
+    @Override
+    public int getRelationalChildCount() {
+        return 1;
+    }
+
+    @Nonnull
+    @Override
+    public List<? extends Quantifier> getQuantifiers() {
+        return ImmutableList.of(inner);
+    }
+
+    @Nonnull
+    @Override
+    public Set<CorrelationIdentifier> getCorrelatedToWithoutChildren() {
+        return ImmutableSet.of();
+    }
+
+    @Nonnull
+    @Override
+    public LogicalUniqueExpression translateCorrelations(@Nonnull final TranslationMap translationMap, @Nonnull final List<? extends Quantifier> translatedQuantifiers) {
+        return new LogicalUniqueExpression(Iterables.getOnlyElement(translatedQuantifiers));
+    }
+
+    @Nonnull
+    @Override
+    public Value getResultValue() {
+        return inner.getFlowedObjectValue();
+    }
+
+    @Override
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
+    public boolean equalsWithoutChildren(@Nonnull final RelationalExpression otherExpression, @Nonnull final AliasMap equivalences) {
+        if (this == otherExpression) {
+            return true;
+        }
+        return getClass() == otherExpression.getClass();
+    }
+
+    @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
+    @Override
+    public boolean equals(final Object other) {
+        return semanticEquals(other);
+    }
+
+    @Override
+    public int hashCodeWithoutChildren() {
+        return 31;
+    }
+
+    @Override
+    public int hashCode() {
+        return semanticHashCode();
+    }
+
+    @Nonnull
+    @Override
+    public PlannerGraph rewriteInternalPlannerGraph(@Nonnull final List<? extends PlannerGraph> childGraphs) {
```

Comment:
This may also be wrong in for `LogicalDistinctExpression` but I don't think we need to rewrite it. Can we just remove this override and then not implement the rest. interface?

Reply:
Done, I thought that ideally, we should a graphical representation of all expressions?

Reply:
We do get a default which is fine here. This interface is just there to allow you to cognate the default.